core/model/identifier.py

```diff
@@ -561,6 +561,26 @@ def _parse_urn(
 
         return cls.for_foreign_id(_db, identifier_type, identifier_string)
 
+    @classmethod
+    @overload
+    def parse_urn(
+        cls,
+        _db: Session,
+        identifier_string: str,
+        must_support_license_pools: bool = False,
+    ) -> tuple[Identifier, bool]:
+        ...
+
+    @classmethod
+    @overload
+    def parse_urn(
+        cls,
+        _db: Session,
+        identifier_string: str | None,
```

Comment:
Should this be just the `str` type, since we've covered the `None` case?

Reply:
@RishiDiwanTT I'm a bit confused by this comment. Do you mean the opposite here? 
> Should this just be `None` since we've covered the `str` case?

We need a type hint for the type union case, that will be used when mypy doesn't know if the type is `None` or `str` statically.

Reply:
Yes, I meant the opposite.